### PR TITLE
Remove a bunch of unused/pointless types.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: server/log/main.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ var storageMap = make(map[int64]storage.LogStorage)
 
 // TODO(Martin2112): Needs to be able to swap out for different storage type
 func simpleMySQLStorageProvider(treeID int64) (storage.LogStorage, error) {
-	return mysql.NewLogStorage(trillian.LogID{[]byte("TODO"), treeID}, *mysqlURIFlag)
+	return mysql.NewLogStorage(treeID, *mysqlURIFlag)
```
Comment (AlCutter) [inline server/log/main.go]: Yes, I thought that too - but I figured we should design/add that when we're ready.

File: storage/tools/common.go [hunk 2]
```diff
@@ -4,22 +4,20 @@ import (
...
 var storageTypeFlag = flag.String("storage_type", "mysql", "Which type of storage to use")
 var mysqlURIFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test",
 	"uri to use with mysql storage")
 var serverPortFlag = flag.Int("port", 8090, "Port to serve log requests on")
 
 // GetLogIDFromFlagsOrDie returns the Trillian LogID from the current flags configuration.
-func GetLogIDFromFlagsOrDie() trillian.LogID {
-	return trillian.LogID{[]byte(*logIDFlag), *treeIDFlag}
+func GetLogIDFromFlagsOrDie() int64 {
+	return *treeIDFlag
```
Comment (Martin2112) [inline storage/tools/common.go]: Yep

File: storage/tools/common.go [hunk 3]
```diff
@@ -29,13 +27,13 @@ func GetLogStorageProviderFromFlags() server.LogStorageProviderFunc {
 	return func(x int64) (storage.LogStorage, error) {
 		// TODO: We need to sort out exactly what a log id is. We really only need a tree id
```
Comment (AlCutter) [inline storage/tools/common.go]: ding'd

File: storage/mysql/log_storage.go [hunk 4]
```diff
@@ -4,16 +4,17 @@ import (
 	"crypto/rand"
 	"crypto/sha256"
 	"database/sql"
+	"encoding/binary"
 	"errors"
 	"fmt"
 
+	"bytes"
```
Comment (AlCutter) [inline storage/mysql/log_storage.go]: weird, fixed.

File: types.go [hunk 5]
```diff
@@ -1,25 +1,14 @@
...
-
-func (h Hash) String() string {
-	return base64.StdEncoding.EncodeToString(h)
-}
-
 // Leaf represents the data behind Merkle leaves.
 type Leaf struct {
 	// MerkleLeafHash is the tree hash of LeafValue
-	MerkleLeafHash Hash
+	MerkleLeafHash []byte
```
Comment (AlCutter) [inline types.go]: gofmt says "you can't have it"

File: hashers.go [hunk 6]
```diff
@@ -22,7 +22,7 @@ func NewHasher(alg HashAlgorithm) (Hasher, error) {
 }
 
 // Digest calculates the digest of b according to the underlying algorithm.
-func (h Hasher) Digest(b []byte) Hash {
+func (h Hasher) Digest(b []byte) []byte {
```
Comment (AlCutter) [inline hashers.go]: yeah, I think it was a bit more readable in places, but the hassle outweighs it.

File: types.go (hunk 5 above)
Comment (Martin2112) [inline types.go]: How tiresome.

**Pull request merged**